lib/vssh/libssh.c

```diff
@@ -549,49 +549,52 @@ static int myssh_is_known(struct Curl_easy *data)
   return rc;
 }
 
-#define MOVE_TO_ERROR_STATE(_r) { \
+#define MOVE_TO_ERROR_STATE(_r) do { \
   state(data, SSH_SESSION_DISCONNECT); \
   sshc->actualcode = _r; \
   rc = SSH_ERROR; \
   break; \
-}
+} while(0)
```

Comment:
This macro now puts the break inside the do-while so it won't break out from the switch like before, changing the flow significantly... I would suggest you remove the break from the macro and write it outside of it in the code - which I think also increases code readability.